src/awkward/_v2/contents/regulararray.py

```diff
@@ -392,14 +398,18 @@ def _getitem_next(self, head, tail, advanced):
                 nextadvanced = ak._v2.index.Index64.empty(
                     self._length * nextsize, self._nplike
                 )
+                # assert (
+                #     nextadvanced.nplike is self._nplike
+                #     and advanced.nplike is self._nplike
+                # )
```

Comment:
More commented-out assertions. I'm getting the impression that these are all in the list-like arrays (ListArray, ListOffsetArray, and RegularArray), and that they all have to do with `array[slicer]`.